lib/goth/application.ex

```diff
@@ -4,17 +4,15 @@ defmodule Goth.Application do
 
   @impl true
   def start(_type, _args) do
-    envs = Application.get_all_env(:goth)
-
-    if envs == [] do
+    if Application.get_all_env(:goth) == [] do
```

Comment:
if app env is empty we're not going to start the legacy supervisor which means this is broken:

```
$ iex
iex> Goth.Token.for_scope("https://www.googleapis.com/auth/cloud-platform")
warning: Goth.Token.for_scope/1 is deprecated. Use Goth.fetch/1 instead
  iex:1

** (exit) exited in: GenServer.call(Goth.TokenStore, {:find, {:default, "https://www.googleapis.com/auth/cloud-platform", nil}}, 5000)
    ** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started
    (elixir 1.14.0-dev) lib/gen_server.ex:1027: GenServer.call/3
    (goth 1.3.0-rc.5) lib/goth/token.ex:365: Goth.Token.for_scope/2
    iex:1: (file)
```

or you wanted to handle that in a separate PR?